lib/document.js

```diff
@@ -580,17 +580,12 @@ Document.prototype._batchSave = function(executeInsert) {
  * @return {Promise}
  */
 Document.prototype._batchSaveSelf = function(executeInsert) {
-  var model = this._getModel();
+  var self = this;
 
   return new Promise(function(resolve, reject) {
-    if (model._tableReady === true) {
+    self.constructor.ready().then(function() {
```

Comment:
We swap protos here: https://github.com/neumino/thinky/blob/master/lib/model.js#L106